### PR TITLE
[TLT-5942][Bugfix] Fix DINOv3 and Visual ChangeNet release regressions

### PR DESCRIPTION
**Pull request by nnagrajrao**
### What changes are proposed in this pull request?

This PR consolidates four TAO 7.2 runtime regressions into two reviewable commits:

- Exposes DINOv3 `train.log_every_n_steps` and forwards it to Lightning, defaulting to every step so short runs publish component and preservation losses.
- Maps DINOv3 SSL architecture tokens to real `BACKBONE_REGISTRY` constructor names in convert guidance.
- Builds the Visual ChangeNet classification criterion from the model architecture so strict checkpoint loading does not depend on irrelevant inference-time training-loss configuration.
- Enforces the ChangeNet loss/architecture pairing only for `fit`.
- Computes conventional FPR and FNR percentages using the actual-pass and actual-defect denominators, with empty-class guards.
- Adds focused regression tests for each behavior, including strict state-dict loading for evaluate, predict, and export configurations.

### Why are the changes needed?

- DINOv3 component losses were logged on steps, but the Trainer cadence was not configurable from the experiment spec; short smoke runs could finish before emitting them.
- DINOv3 conversion printed raw SSL architecture tokens such as `vit_b`, which are not valid downstream registry keys.
- ChangeNet evaluate, infer, and export construct the Lightning module even though they do not use the training loss. Constructor-time loss pairing therefore rejected otherwise valid checkpoints, while simply removing the criterion would break strict loading of `criterion.weight` from CE checkpoints.
- `AOIMetrics` labeled total-sample-normalized error percentages as FPR/FNR. Conventional FPR is `FP / actual_pass` and FNR is `FN / actual_defect`.

### Related issues

- [TLT-5942](https://jirasw.nvidia.com/browse/TLT-5942) / [NVBug 6642016](https://nvbugs.nvidia.com/bug/6642016)
- [TLT-5939](https://jirasw.nvidia.com/browse/TLT-5939) / [NVBug 6642007](https://nvbugs.nvidia.com/bug/6642007)
- [TLT-5925](https://jirasw.nvidia.com/browse/TLT-5925) / [NVBug 6604829](https://nvbugs.nvidia.com/bug/6604829)
- [TLT-5923](https://jirasw.nvidia.com/browse/TLT-5923) / [NVBug 6603000](https://nvbugs.nvidia.com/bug/6603000)
- Schema-source dependency: [NVIDIA-TAO/tao-core#35](https://github.com/NVIDIA-TAO/tao-core/pull/35)
- TensorRT metric-parity follow-up: [NVIDIA-TAO/tao-deploy#41](https://github.com/NVIDIA-TAO/tao-deploy/issues/41)

### Does this PR introduce _any_ user-facing change?

Yes.

- DINOv3 training logs every step by default and accepts `train.log_every_n_steps` overrides.
- DINOv3 conversion prints a directly usable downstream registry key.
- ChangeNet non-training actions no longer require a matching `train.classify.loss` value.
- Reported `test_fpr` and `test_fnr` are class-conditional percentages rather than total-dataset percentages.

### How was this patch tested?

Using the exact ARM64 image pinned in `docker/manifest.json`:

```text
tests/ssl_unit_test/dinov3/test_config.py
tests/cv_unit_test/optical_inspection/test_metrics.py

20 passed
```

```text
tests/ssl_unit_test/dinov3/test_backbone_v2_interop.py -k 'not registry'

5 passed, 1 skipped (GPU/weights), 1 deselected
```

The configured static checks passed:

```text
license header: Passed
pylint: Passed
pydocstyle: Passed
flake8: Passed
git diff --check: Passed
Python compilation: Passed
DCO sign-off: Passed for both commits
```

The remaining train/convert/registry/ChangeNet tests import the encrypted `eff_tao_encryption` ARM extension, which exits with `Illegal instruction` under Colima before pytest can execute assertions. Hosted `/build` CI is requested below on the supported runner. The TruffleHog hook's fresh Go environment also exceeded the local Colima memory limit; hosted CI runs the repository gate.

### Was this patch authored or co-authored using generative AI tooling?

Generated-by: OpenAI Codex (GPT-5)

### Release note

DINOv3 now exposes reliable short-run logging and valid conversion guidance, while Visual ChangeNet supports checkpoint-safe non-training actions and reports conventional class-conditional FPR/FNR percentages. Existing baselines and dashboards keyed on `val_fpr`, `test_fpr`, or `test_fnr` will generally shift upward because the denominator shrinks from all samples to the relevant ground-truth class; values remain percentages.

### Checklist

- [X] My commits are signed off per the DCO (`git commit -s`) — see `CONTRIBUTING.md`
- [X] I have read the contributing guidelines
- [X] The code follows the project's style, and lint and format checks pass locally
- [X] I added or updated tests covering this change
- [ ] All tests pass locally; ARM-native tests are delegated to `/build`
- [X] I added or updated documentation (config metadata, messages, and docstrings)
- [ ] I updated the version, changelog, or migration notes if this change requires it
- [ ] If this touches components that are optional to install, the imports are guarded
      so the package still works without them (reviewers: please verify this)
- [X] No secrets, credentials, proprietary data, or customer data are included in this PR
- [X] I understand this contribution is licensed under the repository's license, and that
      my commit author name and email become permanently public once merged

### Notes for reviewers

Review commit `0db9ddd` for DINOv3 and `a52af74` for ChangeNet. The ChangeNet criterion remains architecture-derived so CE checkpoint state (`criterion.weight`) loads strictly; only the requested training loss validation moves to `setup("fit")`. For TLT-5923, the filed values were already percentages, not raw counts; this patch corrects their denominators and retains percentage units. The matching tao-deploy change is tracked in [issue #41](https://github.com/NVIDIA-TAO/tao-deploy/issues/41).

### CODE REVIEW

File: nvidia_tao_pytorch/cv/optical_inspection/model/build_nn_model.py [hunk 1]
```diff
@@ -237,11 +237,13 @@ def compute(self):
...
+        zero = torch.zeros_like(total, dtype=torch.float32)
         metric_collect = {}
-        metric_collect['total_accuracy'] = ((self.match_pass + self.match_fail) / (self.tot_pass + self.tot_fail)) * 100
-        metric_collect['defect_accuracy'] = torch.Tensor([0]) if self.tot_fail == 0 else (self.match_fail / self.tot_fail) * 100
-        metric_collect['false_alarm'] = (self.mismatch_pass / (self.tot_pass + self.tot_fail)) * 100
-        metric_collect['false_negative'] = (self.mismatch_fail / (self.tot_pass + self.tot_fail)) * 100
+        metric_collect['total_accuracy'] = zero if total == 0 else ((self.match_pass + self.match_fail) / total) * 100
+        metric_collect['defect_accuracy'] = zero if self.tot_fail == 0 else (self.match_fail / self.tot_fail) * 100
+        metric_collect['false_alarm'] = zero if self.tot_pass == 0 else (self.mismatch_pass / self.tot_pass) * 100
+        metric_collect['false_negative'] = zero if self.tot_fail == 0 else (self.mismatch_fail / self.tot_fail) * 100
```
Comment (vpraveen-nv) [inline nvidia_tao_pytorch/cv/optical_inspection/model/build_nn_model.py:246]: Cross-project parity gap: tao-deploy carries duplicated copies of this metric with the old total-sample denominators — `nvidia_tao_deploy/cv/visual_changenet/classification/utils.py` (L61-62) and the optical_inspection evaluate path. After this lands, `<model> evaluate` in the PyTorch container and TRT-engine evaluate in tao-deploy will report different FPR/FNR for the same checkpoint/dataset. Needs a matching tao-deploy PR (or a tracked follow-up bug) so the two containers agree.

Also worth noting in the release note: existing baselines/dashboards keyed on `val_fpr`/`test_fpr`/`test_fnr` will shift upward (denominator shrinks) even though ordering across checkpoints is preserved for a fixed validation split.

File: nvidia_tao_pytorch/cv/visual_changenet/classification/models/cn_pl_model.py [hunk 2]
```diff
@@ -83,26 +83,40 @@ def _build_model(self, export):
...
+                raise ValueError(
+                    "Class weights must be provided for each class: "
+                    f"received {len(cls_weight)} weights for {n_class} classes"
+                )
             self.class_weights = torch.tensor(cls_weight)
             self.criterion = nn.CrossEntropyLoss(weight=self.class_weights)
-        else:
-            raise NotImplementedError(f"loss function {self.loss_fn} is not implemented")
+
+    def setup(self, stage):
```
Comment (vpraveen-nv) [inline nvidia_tao_pytorch/cv/visual_changenet/classification/models/cn_pl_model.py:107]: Nit: call `super().setup(stage)` before the early return. `pl.LightningModule.setup` is a no-op today, but if `TAOLightningModule` ever adds setup behavior, this override would silently swallow it for all non-fit stages.

**Review by vpraveen-nv — APPROVED** [review]
minor nits. approved.

File: nvidia_tao_pytorch/cv/visual_changenet/classification/models/cn_pl_model.py (hunk 2 above)
Comment (nnagrajrao) [reply nvidia_tao_pytorch/cv/visual_changenet/classification/models/cn_pl_model.py:107]: Fixed in 2ebef75 by calling super().setup(stage) before the stage guard. The complete committed-range pre-commit suite passes.

File: nvidia_tao_pytorch/cv/optical_inspection/model/build_nn_model.py (hunk 1 above)
Comment (nnagrajrao) [reply nvidia_tao_pytorch/cv/optical_inspection/model/build_nn_model.py:246]: Valid. The matching TensorRT-side behavior is now tracked in NVIDIA-TAO/tao-deploy#41, which is linked from Related issues and Notes for reviewers. I also updated the release note to call out the expected upward shift for val_fpr, test_fpr, and test_fnr while retaining percentage units.
